src/puppetlabs/puppetdb/scf/hash_debug.clj

```diff
@@ -35,63 +36,45 @@
   "Updates the edge relationships from the DB to keywords for easier diffing
    with the new catalog."
   [edges]
-  (map #(update-in % [:relationship] keyword) edges))
+  (map #(update % :relationship keyword) edges))
 
 (defn resource-params->kwds
   "Updates the resource parameter keys from the DB to keywords for easier diffing
    with the new catalog."
   [resources]
-  (map (fn [resource]
-         (update-in resource [:parameters] #(kitchensink/mapkeys keyword %)))
-       resources))
+  (map #(update % :parameters (partial kitchensink/mapkeys keyword)) resources))
 
-(defn query-resources
-  [certname
-   version]
-  (let [query-sql (resources/query->sql version ["=" "certname" certname])
-        results (resources/query-resources version query-sql "")]
-    (:result results)))
+(defn get-entity
+  "An unfortunate copy of `query_eng.clj`'s `stream-query-result` without the `with-transacted-connection`
```

Comment:
Is with-transacted-connection not re-entrant? It seems like it should be fine to just pass *db* down to stream-query result.